nvdimm: support nvdimm on arm64 kernel

Pennyzct opened the pull request:
let's open nvdimm-related kernel config parameters on arm64, such as
`CONFIG_ACPI_NFIT`, etc. and we also need to backport patch [kvm:arm64:Dynamic IPA and 52bit IPA]( https://patchwork.kernel.org/cover/10616271/) and related dependency into v4.19.X to fully support nvdimm from guest kernel.
Former patch has already been merged into v4.20.X.

Fixes: #376

Signed-off-by: Penny Zheng <penny.zheng@arm.com>

Pennyzct commented:
/test

grahamwhaley commented on kernel/configs/arm64_kata_kvm_4.19.x:
```diff
@@ -1141,6 +1175,13 @@ CONFIG_OF_NET=y
 # CONFIG_OF_OVERLAY is not set
 # CONFIG_PARPORT is not set
+CONFIG_PNP=y
+CONFIG_PNP_DEBUG_MESSAGES=y
```
not sure if you set this deliberately? Payoff of size vs debuggability...

Pennyzct replied:
disable it. ;)

grahamwhaley commented on kernel/configs/arm64_kata_kvm_4.19.x:
```diff
@@ -1533,7 +1576,26 @@ CONFIG_POWER_SUPPLY=y
+# CONFIG_THERMAL_GOV_FAIR_SHARE is not set
+# CONFIG_THERMAL_GOV_STEP_WISE is not set
+# CONFIG_THERMAL_GOV_BANG_BANG is not set
+CONFIG_THERMAL_GOV_USER_SPACE=y
```
On purpose? Maybe you gained some default `=y` items during a new config generation. If you have not already, you might want to scrutinize the diff for any stray new additions you didn't intend.

Pennyzct replied:
noooo, my bad, I just quickly compared the new diff with x86 config(`x86_64_kata_kvm_4.19.X`) to make two configs as same as possible, not scrutinizing what these diff represent.
I will turn off the CONFIG_THERMAL, since VM may not be involved in thermal managment. ;)

grahamwhaley commented:
@Pennyzct - no, no way to trigger the ARM CI here - in fact, we tried to add the Power8 CI here last week iirc (or maybe that was osbuilder), and it didn't work as it went to pull OBS images that did not exist I think.
But, @chavafg , might there be a way to trigger a PR on another repo that has this PR as a dependency, to get it pulled in? I'm not sure..

grahamwhaley commented:
ooh, 'leaky pods' on the 18.04 CI....
@nitkon - did you get any further on that, or are you now needing direction/assistance?

```
bash sanity/check_sanity.sh
ERROR: 3 pods left and found at /var/lib/vc/sbs
Makefile:45: recipe for target 'functional' failed
make: *** [functional] Error 1
Build step 'Execute shell' marked build as failure
```

nitkon commented:
Yes, @jodh-intel suggested to get a kernel backtrace of such VM Panics. We can probably get that by enabling serial console and then redirecting the guest serial to some host file. 
However reproducing this issue locally would be the first challenge as running these tests individually don't result in panic(In my testing atleast)

jodh-intel commented:
@nitkon - you should get something in the proxy log without the need for a serial console. Certainly oops's are logged an I've seen panics when the rootfs is not mountable.

Pennyzct commented:
Hi~ @grahamwhaley  I had another relevant [PR](https://github.com/kata-containers/runtime/pull/1323) in runtime repo, and i will try to make this pr as dependency to see if ARM CI on runtime repo could test these two PRs together. ;)

Pennyzct commented:
Hi~@grahamwhaley  I have addressed the comments and done the fix, ptal. ;)
Also, I have already done the re-base and update the `kata_config_version`.
For now, I'm struggling to add this PR as dependency for another related PR [runtime/#1323](https://github.com/kata-containers/runtime/pull/1323). 
So I could trigger the ARM CI under` runtime/` repo to test this two PRs together. I definitely once saw how to make this happen, like `dependency:  packaging/#377`, I just couldn't find the reference, would anyone like to lend a hand here? ;) @jodh-intel @chavafg

jodh-intel commented:
Hi @Pennyzct - It's actually documented here:

- https://github.com/kata-containers/tests#breaking-compatibility

And here's an example of a PR that used it:

- https://github.com/kata-containers/runtime/pull/980

Pennyzct commented:
Hi~ @jodh-intel tons of thanks!!! 
I have already add dependencies in PR [runtime/#1323](https://github.com/kata-containers/runtime/pull/1323),  let's see how it works~~~~;)

amshinde commented:
/test

amshinde commented:
Still seing failures:
```
Kernel directory has changes check kernel/kata_config_version changed
ERROR: Please bump version in kernel/kata_config_version
```
In spite of the version being changed. Retriggering one more time...

/test

Pennyzct commented:
Hi~ @amshinde  re-based and updated the version number.

jodh-intel commented:
Thanks @Pennyzct. Let's retry the CI...

/retest
